### ISSUE TITLE
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 response listeners added.

### ISSUE THREAD

**Issue by misozask** [jamen/pull-fetch#8]
Hi,
it seems you have a bug here:     

```
  function response () {
        const source = (end, cb) => {
          if (end) return cb(end)
            req.on('response', res => {  // <=here
              res.on('data', chunk => cb(null, chunk))
              res.on('end', () => cb(true))
            res.on('error', err => cb(err))
          })
        }
```
On each iteration you are adding new 'response' listener.
Should be enough to add it once, e.g:
```
      function response () {
        let registered = false;
        const source = (end, cb) => {
          if (end) return cb(end)
          if(!registered){
            req.on('response', res => {
              res.on('data', chunk => {
                cb(null, chunk)
              })
              res.on('end', () => cb(true))
              res.on('error', err => cb(err))
            })
            registered = true;
          }
        }
```

**Comment by misozask** [comment]
One more thing, when I want to abort stream, e.g. after first chunk:
```
drain(x => {
        console.log(x.toString());
        return false;
    })(response.source);
```
it will keep processing chunks.
Seems req.abort() might help:
```
        const source = (end, cb) => {
          if (end) {
            req.abort();
            return cb(end)
          }
```
Or am I missing something here?

**Comment by jamen** [comment]
Both seem like good improvements.  Feel free to send a PR, or I can get around to it sometime this week. :smile:

**Comment by misozask** [comment]
I'll try tomorrow...
One more thing - was this an intention, to make callback on every 'data' event? :
 res.on('data', chunk => {
    cb(null, chunk)
 })
I guess its against the pull-stream principles. 'Sink' should pull the chunks from 'source'.
But now it looks more like pushing chunks to sink .... 

I'll try to improve it with buffer, in a similar way as it is in https://github.com/pull-stream/pull-ws/blob/master/source.js

**Comment by jamen** [comment]
You're right, that is another mistake. Each read should give one chunk, but should not give multiple chunks on a single read.

**Comment by misozask** [comment]
Can you please give me permission to push new branch + PR? Thanks

**Comment by jamen** [comment]
Yeah. Thanks for your help. Sent an invite: https://github.com/jamen/pull-fetch/invitations

**Comment by jamen** [comment]
Published `pull-fetch@1.1.0` with your changes.  Thanks!
